### ISSUE TITLE
`getFeaturesInExtent` not returning features for tile layer when zoom is outside `minZoom` or `maxZoom`

### ISSUE THREAD

**Issue by Jesse97** [openlayers/openlayers#16189]
**Describe the bug**
When zooming a few levels before `minZoom` or past `maxZoom` of a `VectorTileLayer`, `getFeaturesInExtent` doesn't return visible features anymore. This still worked in OpenLayers 9.2.4.
When digging into the issue, it looks like that this was caused by changes in #15905 and/or #16007, as the implementation of `getFeaturesInExtent` differs between the old and new versions.

**To reproduce**
Forking from the [OGC Vector Tiles example](https://openlayers.org/en/latest/examples/ogc-vector-tiles.html), this reproduction presents the results of `getFeaturesInExtent` on top of the displayed tile layer. The example tileset has zoom levels 0-7 (implicitly setting `minZoom` and `maxZoom` to that range). Getting features yields results on levels 8 and 9 as well when zooming in, but doesn't return anything from level 10 onwards. After zooming back out to level 7, features are returned again.

```js
import {Feature} from 'ol';
import MVT from 'ol/format/MVT.js';
import VectorLayer from 'ol/layer/Vector';
import VectorTileLayer from 'ol/layer/VectorTile';
import Map from 'ol/Map.js';
import OGCVectorTile from 'ol/source/OGCVectorTile';
import VectorSource from 'ol/source/Vector';
import View from 'ol/View';

const vectorTileSource = new OGCVectorTile({
    url: 'https://maps.gnosis.earth/ogcapi/collections/NaturalEarth:cultural:ne_10m_admin_0_countries/tiles/WebMercatorQuad',
    format: new MVT({featureClass: Feature}),
    wrapX: false
});

const vectorTileLayer = new VectorTileLayer({
    source: vectorTileSource,
    style: {
        'stroke-width': 0.6,
        'stroke-color': '#8c8b8b',
        'fill-color': '#f7f7e9'
    }
});

const debugLayer = new VectorLayer({
    source: new VectorSource(),
    style: {
        'stroke-width': 1,
        'stroke-color': '#f00',
        'stroke-line-dash': [5, 5],
        'fill-color': '#f002'
    }
});

const map = new Map({
    target: 'map',
    layers: [
        vectorTileLayer,
        debugLayer
    ],
    view: new View({
        center: [0, 0],
        zoom: 0,
        multiWorld: false
    })
});

map.on('moveend', () => {
    console.log('zoom', map.getView().getZoom());
    const features = vectorTileLayer.getFeaturesInExtent(map.getView().calculateExtent());
    debugLayer.getSource().clear();
    debugLayer.getSource().addFeatures(features.map(f => new Feature(f.getGeometry())));
});
```

**Expected behavior**
Even with a `minZoom` or `maxZoom` applied on the source of a `VectorTileLayer`, I expect features to be returned by `getFeaturesInExtent`, as they are still visibly rendered on the map.

**Comment by ahocevar** [comment]
Thanks for reporting this, will be fixed with #16190.

**Comment by Jesse97** [comment]
Many thanks!
